genesyscloud/telephony_provider_edges_trunkbasesettings/resource_genesyscloud_telephony_providers_edges_trunkbasesettings.go

```diff
@@ -212,33 +154,45 @@ func updateTrunkBaseSettings(ctx context.Context, d *schema.ResourceData, meta i
 	trunkBase.Version = trunkBaseSettings.Version
 
 	log.Printf("Updating trunk base settings %s", name)
-	trunkBaseSettings, resp, err := edgesAPI.PutTelephonyProvidersEdgesTrunkbasesetting(d.Id(), trunkBase)
+	trunkBaseSettings, resp, err := proxy.UpdateTrunkBaseSetting(ctx, d.Id(), trunkBase)
 	if err != nil {
 		return util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to update trunk base settings %s error: %s", d.Id(), err), resp)
 	}
 
-	log.Printf("Updated trunk base settings %s", *trunkBaseSettings.Id)
+	if trunkBaseSettings != nil && trunkBaseSettings.Id != nil {
+		log.Printf("Updated trunk base settings %s", *trunkBaseSettings.Id)
+	} else {
+		log.Printf("Error: trunkBaseSettings or its Id is nil")
+	}
 
 	return readTrunkBaseSettings(ctx, d, meta)
 }
 
 func readTrunkBaseSettings(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
-	edgesAPI := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(sdkConfig)
+	proxy := getTrunkBaseSettingProxy(sdkConfig)
+
 	cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceTrunkBaseSettings(), constants.DefaultConsistencyChecks, resourceName)
 
 	log.Printf("Reading trunk base settings %s", d.Id())
 	return util.WithRetriesForRead(ctx, d, func() *retry.RetryError {
-		trunkBaseSettings, resp, getErr := edgesAPI.GetTelephonyProvidersEdgesTrunkbasesetting(d.Id(), true)
+		trunkBaseSettings, resp, getErr := proxy.GetTrunkBaseSettingById(ctx, d.Id())
+
 		if getErr != nil {
 			if util.IsStatus404(resp) {
 				return retry.RetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("Failed to read trunk base settings %s | error: %s", d.Id(), getErr), resp))
 			}
 			return retry.NonRetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("Failed to read trunk base settings %s | error: %s", d.Id(), getErr), resp))
 		}
 
-		d.Set("name", *trunkBaseSettings.Name)
-		d.Set("state", *trunkBaseSettings.State)
+		if trunkBaseSettings != nil && trunkBaseSettings.Name != nil {
```

Comment:
Don't we have a helper utility functions that can check this before setting the data state?

Reply:
I dont think we have one for checking the core object and the child object.